homeassistant/components/ecovacs/sensor.py

```diff
@@ -0,0 +1,269 @@
+    """Add entities for passed config_entry in HA."""
+    controller: EcovacsController = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities: list[EcovacsEntity] = get_supported_entitites(
+        controller, EcovacsSensor, ENTITY_DESCRIPTIONS
+    )
+    for device in controller.devices:
+        lifespan_capability = device.capabilities.life_span
+        for component in _SUPPORTED_LIFE_SPAN_TYPES:
+            if component in lifespan_capability.types:
+                entities.append(
+                    EcovacsLifeSpanSensor(device, lifespan_capability, component)
+                )
+
+        if capability := device.capabilities.error:
+            entities.append(EcovacsLastErrorSensor(device, capability))
+
+    async_add_entities(entities)
+
+
+class EcovacsSensor(
+    EcovacsDescriptionEntity[CapabilityEvent],
+    SensorEntity,
+):
+    """Ecovacs sensor."""
+
+    entity_description: EcovacsSensorEntityDescription
+
+    async def async_added_to_hass(self) -> None:
+        """Set up the event listeners now that hass is ready."""
+        await super().async_added_to_hass()
+
+        async def on_event(event: Event) -> None:
+            value = self.entity_description.value_fn(event)
+            if value is None:
+                return
+
+            self._attr_native_value = value
+            self.async_write_ha_state()
+
+        self._subscribe(self._capability.event, on_event)
+
+
+class EcovacsLifeSpanSensor(
+    EcovacsDescriptionEntity[CapabilityLifeSpan],
+    SensorEntity,
+):
+    """Life span sensor."""
+
+    entity_description: SensorEntityDescription
+
+    def __init__(
+        self,
+        device: Device,
+        capability: CapabilityLifeSpan,
+        component: LifeSpan,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize entity."""
+        key = f"life_span_{component.name.lower()}"
+        super().__init__(
+            device,
+            capability,
+            SensorEntityDescription(
+                key=key,
+                translation_key=key,
+                native_unit_of_measurement="%",
+                entity_category=EntityCategory.DIAGNOSTIC,
+            ),
+            **kwargs,
+        )
+        self._component = component
+
+    async def async_added_to_hass(self) -> None:
+        """Set up the event listeners now that hass is ready."""
+        await super().async_added_to_hass()
+
+        async def on_event(event: LifeSpanEvent) -> None:
+            if event.type == self._component:
+                self._attr_native_value = event.percent
+                self._attr_extra_state_attributes = {
+                    "remaining": floor(event.remaining / 60)
+                }
+                self.async_write_ha_state()
+
+        self._subscribe(self._capability.event, on_event)
+
+
+class EcovacsLastErrorSensor(
+    EcovacsEntity[CapabilityEvent[ErrorEvent]],
+    SensorEntity,
+):
+    """Last error sensor."""
+
+    _always_available: bool = True
```

Comment:
This is already typed as bool right?
```suggestion
    _always_available = True
```